// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package research.simulation;

// Defines types of turbulent kinetic energy (TKE) models with required
// parameters.
message TKE {
  // Sets the TKE to a constant.
  message Constant {
    optional float tke_constant = 1 [default = 50.0];
  }

  // Computes the TKE based on flow field variables algebraically.
  message Algebraic {}

  // Derives the TKE from the turbulent viscosity.
  message TurbulentViscosity {}

  oneof tke_model_option {
    Constant constant = 1;
    Algebraic algebraic = 2;
    TurbulentViscosity turbulent_viscosity = 3;
  }
}
